lib/model.js

```diff
@@ -213,26 +214,32 @@ var Events = require('./events'),
         validate: function () {
             var model = this;
             return model.ready().then(function () {
-                return Vow.all(model._attributesAr.map(function (attribute) {
+                return Vow.allResolved(model._attributesAr.map(function (attribute) {
                     return attribute.validate();
                 }));
-            }).then(function (validateResults) {
-                var isValid = validateResults.every(function(result) {
-                    return result === true;
+            }).then(function (validationResults) {
+                var errors = [],
+                    error;
+
+                validationResults.forEach(function (validationResult, index) {
+                    var error;
+                    if (Vow.isPromise(validationResult)) {
+                        error = validationResult.isRejected() && validationResult.valueOf();
+                    } else {
+                        error = validationResult instanceof Error && validationResult;
+                    }
+                    if (error) {
+                        error.attribute = model._attributesAr[index];
+                        errors.push(error);
+                    }
                 });
 
-                if (isValid) {
+                if (!errors.length) {
```

Comment:
Avoid using `if (!..) else`, it always might me written in direct way: `if (..) else`. In that case it looks more convenient and clear.